fabric8-maven-plugin-build/src/main/java/org/fabric8/maven/plugin/build/Fabric8MavenPluginResourceGeneratorBuilder.java

```diff
@@ -0,0 +1,126 @@
+    private String namespace;
+    private boolean mvnDebugOutput;
+    private boolean quietMode;
+    private String[] profiles = new String[0];
+    private Map<String, String> properties = new HashMap<>();
+
+    public Fabric8MavenPluginResourceGeneratorBuilder namespace(String namespace) {
+        this.namespace = namespace;
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder goals(String[] goals) {
+        this.goals = goals;
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder pluginConfigurationIn(Path pom) {
+        this.pom = pom;
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder quiet() {
+        return quiet(true);
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder quiet(boolean quiet) {
+        this.quietMode = quiet;
+        return this;
+    }
+
+    /**
+     * Enables mvn debug output (-X) flag. Implies build logging output.
+     */
+    private Fabric8MavenPluginResourceGeneratorBuilder withDebugOutput(boolean debug) {
+        this.mvnDebugOutput = debug;
+        quiet(!debug);
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder debug(boolean debug) {
+        this.mvnDebugOutput = debug;
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder profiles(List<String> profiles) {
+        return profiles(profiles.toArray(new String[profiles.size()]));
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder profiles(String... profiles) {
+        this.profiles = profiles;
+        return this;
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder withProperties(List<String> propertiesPairs) {
+        return withProperties(propertiesPairs.toArray(new String[propertiesPairs.size()]));
+    }
+
+    public Fabric8MavenPluginResourceGeneratorBuilder withProperties(String... propertiesPairs) {
+        if (propertiesPairs.length % 2 != 0) {
+            throw new IllegalArgumentException("Expecting even amount of variable name - value pairs to be passed. Got "
+                + propertiesPairs.length
+                + " entries. "
+                + Arrays.toString(propertiesPairs));
+        }
+
+        for (int i = 0; i < propertiesPairs.length; i += 2) {
+            this.properties.put(propertiesPairs[i], propertiesPairs[i + 1]);
+        }
+
+        return this;
+    }
+
+    public void build() {
+        final ConfigurationDistributionStage distributionStage = EmbeddedMaven
+            .forProject(pom.toFile())
+            .setQuiet(quietMode)
+            .useDefaultDistribution()
+            .setDebug(mvnDebugOutput)
+            .setDebugLoggerLevel()
+            .setGoals(goals)
+            .addProperty("fabric8.namespace", namespace);
+
+        //Remove this after fixing issue  https://github.com/eclipse/che/issues/9105
```

Comment:
If we intend to merge it before the issue in Che is fixed, please mark it with `TODO` or `FIXME` and create issue in Cube so we won't forget about it.